pkg/cloudprovider/providers/vsphere/vsphere.go

```diff
@@ -1288,7 +1288,9 @@ func (vs *VSphere) NodeAdded(obj interface{}) {
 	}
 
 	klog.V(4).Infof("Node added: %+v", node)
-	vs.nodeManager.RegisterNode(node)
+	if err := vs.nodeManager.RegisterNode(node); err != nil {
```

Comment:
Note that `RegisterNode` always returns nil, as does `UnRegisterNode`.  Looks like those methods could be changed to have no return value, instead of unused `error`.

Reply:
Initially that was true, but `RegisterNode` calls `DiscoverNode` and that does return an error which has been ignored so far. I've changed that above.